src/mz/src/bin/mz/upgrader.rs

```diff
@@ -0,0 +1,289 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Command-line version checker.
+
+use axum::http::HeaderValue;
+use hyper::{
+    header::{ACCEPT, USER_AGENT},
+    HeaderMap,
+};
+use mz::{error::Error, ui::OutputFormatter, VERSION};
+use mz_build_info::build_info;
+use serde::Deserialize;
+
+use std::io::Write;
+use std::path::PathBuf;
+use std::time::{SystemTime, UNIX_EPOCH};
+use std::{env, path::Path};
+use std::{
+    fs::{File, OpenOptions},
+    io::{BufRead, BufReader},
+};
+
+/// This const variable represents a whole week in secs =
+/// (7 days * 24 hours * 60 minutes * 60 seconds)
+/// Chronos lib could fit very well here but I avoided
+/// to add an additional lib to the crate.
+const SECONDS_IN_A_WEEK: u64 = 7 * 24 * 60 * 60;
+const TEMP_FILE_NAME: &str = ".mz.ver";
```

Comment:
Using a temporary file for this is probably not right! Temporary files are oten cleared on restart, and you explicitly _don't_ want this going away when the computer restarts.

I think what you want here is the cache dir: https://docs.rs/dirs/latest/dirs/fn.cache_dir.html